js/source/geojson_source.js

```diff
@@ -1,13 +1,27 @@
 'use strict';
 
 var util = require('../util/util');
-var Evented = require('../util/evented');
-var TilePyramid = require('./tile_pyramid');
-var Source = require('./source');
 var urlResolve = require('resolve-url');
 var EXTENT = require('../data/bucket').EXTENT;
 
-module.exports = GeoJSONSource;
+var webworkify = require('webworkify');
+
+module.exports.create = function (id, options, dispatcher, onChange, callback) {
+    var source = new GeoJSONSource(id, options, dispatcher, onChange);
+
+    //TODO: is it to aggressive to do this on source creation, as opposed to
+    //lazily at the first tile request?
```

Comment:
I don't think so. Be aggressive! Given the choice between using less bandwidth and showing users their data faster, I'd choose the latter.